lib/rabbit_feed/connection.rb

```diff
@@ -0,0 +1,22 @@
+module RabbitFeed
+  class Connection
+    include Singleton
+
+    def initialize
+      @connection = Bunny.new RabbitFeed.configuration.connection_options
+      @connection.start
+      @channel = @connection.create_channel
+      @mutex = Mutex.new
+    end
+
+    protected
+
+    attr_reader :channel
+
+    def thread_safe &block
```

Comment:
> Shouldn't this be handled with something like (but better as this one still has a tiny window where it can return false for 2 threads)

I agree that would be helpful. Added in bec0c5f